Add Python to Language Matrix

JacksonWeber opened the pull request:
Add Python to language matrix.

JacksonWeber committed:
Add python to language matrix

TimothyMothra commented:
I can't tell if this change will conflict with this line in the other file:

https://github.com/microsoft/ApplicationInsights-node.js-native-metrics/blob/d149327c9fecaa1a117e9a3ee3b62b4dc3e75270/.github/workflows/codeql-analysis.yml#L32-L35

I'm not an expert on the codeQL configuration, but if this was my repo I would merge these two files to keep it simple.

I'm fine with merging this change and waiting a few days to see if the SFI task resolves. 
If the task doesn't resolve, I want to set the expectation that we may need further changes.

JacksonWeber commented:
> I can't tell if this change will conflict with this line in the other file:
> 
> https://github.com/microsoft/ApplicationInsights-node.js-native-metrics/blob/d149327c9fecaa1a117e9a3ee3b62b4dc3e75270/.github/workflows/codeql-analysis.yml#L32-L35
> 
> I'm not an expert on the codeQL configuration, but if this was my repo I would merge these two files to keep it simple.
> 
> I'm fine with merging this change and waiting a few days to see if the SFI task resolves. If the task doesn't resolve, I want to set the expectation that we may need further changes.

It almost certainly will, sorry I added this to the wrong configuration file, I'll update that now.

TimothyMothra commented:
> It almost certainly will, sorry I added this to the wrong configuration file, I'll update that now.

no worries, we're learning together. Thanks for making the change

JacksonWeber committed:
Update CodeQL config and analysis languages

JacksonWeber commented:
> > It almost certainly will, sorry I added this to the wrong configuration file, I'll update that now.
> 
> no worries, we're learning together. Thanks for making the change

Updated and ready for re-review

TimothyMothra commented:
haha, the CodeQL workflow is failing:

> Exit code was 32 and last log line was: CodeQL detected code written in C/C++ and JavaScript/TypeScript, but not any written in Python. Confirm that there is some source code for Python in the project. For more information, review our troubleshooting guide at 

Is it possible that you're excluding the directory?

JacksonWeber committed:
Remove '*.gyp' from CodeQL paths-ignore
